src/components/internal/Popover.tsx

```diff
@@ -11,17 +12,24 @@ interface PopoverProps extends React.PropsWithChildren<any> {
 
 /** Popover is used for generating menus and list-boxes */
 export function Popover(props: PopoverProps) {
-  const { popoverRef, positionProps, width, children, onClose, isOpen } = props;
+  const { triggerRef, popoverRef, positionProps, width, children, onClose, isOpen } = props;
   const { overlayProps } = useOverlay(
     {
       onClose,
       isOpen,
       isDismissable: true,
-      shouldCloseOnInteractOutside: () => {
+      shouldCloseOnInteractOutside: (e) => {
         // By default when passing `isDismissable: true` then Popover will `stopPropagation` of the PointerDown event, which is used nearly everywhere in React-Aria (like for clicking buttons)
         // We do not want that propagation to be stopped, but we still want the overlay to be dismissable.
         // When providing `isDimissable: true`, then you can also provide this callback function, `shouldCloseOnInteractOutside`
         // By returning `false` in this function it will no longer call `stopPropagation`, but it also will not call `onHide` for us, so we need to call `onClose` ourselves.
+
+        // If we clicked the trigger element (or within it), then that will call a `state.toggle` for us.
+        // Return early if that happens, otherwise we'd call `onClose`, then the trigger would toggle it back open.
+        if (triggerRef.current?.contains(e)) {
```

Comment:
Getting to the point where would love to have some tests for these things; seems likely to regress as we poke/fight various react-aria things. Don't have to block the PR on it if this is fixing a prod bug.

Reply:
Granted, maybe this is past jsdom-testable-behavior? Not sure.

Reply:
Maybe, but regardless I agree. We need to start having more testing around our Beam components.